### PR TITLE
Add basic module for org-roam

### PR DESCRIPTION
**Pull request by jethrokuan**
This is just a reference module for Org-roam, not sure if it's fully appropriate. #2759

### CODE REVIEW

**Comment by hlissner** [comment]
Hey! I've been meaning to add a `+roam` flag, but haven't found time to look into it yet. I have a few issues with this PR, however:

1. It initializes org-roam-mode at startup, via `:hook (after-init . org-roam-mode)`. This pulls in all of org; a very expensive web of packages to load. Can this be avoided?
2. It binds its own leader keys. This should be relegated to the `:config default` module. Modules should only bind their own localleader keys.
3. Since I [lazy load org-protocol](https://github.com/hlissner/doom-emacs/blob/develop/modules/lang/org/config.el#L680-L710), it should be safe to lazy load org-roam-protocol separately, e.g.

```elisp
(use-package! org-roam-protocol
  :after org-protocol)
```

**Comment by hlissner** [comment]
Oh, forgot to mention, could you also add `+roam` to the list of `:lang org`'s flags in a) [the module's README.org](https://github.com/hlissner/doom-emacs/blob/develop/modules/lang/org/README.org#module-flags) (in alphabetical order) under description > module flags and b) in [docs/modules.org](https://github.com/hlissner/doom-emacs/blob/develop/docs/modules.org#lang)

**Comment by jethrokuan** [comment]
Ah no problem, I just copied whatever I had in my config over as a reference. This is especially since there is little information about the company backend for org-roam. It might be easier if you made a separate PR.

1. Yes, but the mode needs to be on for the rest of the functions to work properly. Calling the interactive functions don't load the mode, so some patching would need to be done here.

2. I see. I'm not using the evil mode, so the mapping stuff are still a bit confusing to me.

3. Ah okay!

**Comment by Wetlize** [comment]
> Yes, but the mode needs to be on for the rest of the functions to work properly. Calling the interactive functions don't load the mode, so some patching would need to be done here.

Just place `(org-roam-mode +1)` at the end of `:config` section and it should activate the mode on the actual demand (i.e. when any of the autoloaded `:commands` will be called).

**Comment by hlissner** [comment]
As of f28a304 the lang/org module now has a `+roam` flag, and should be sufficiently lazy loaded without compromising it. Let me know if that isn't the case and I'll have another look at it. Thanks for bringing it to my attention!

File: modules/lang/org/contrib/roam.el [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        :desc "org-roam-insert" "i" #'org-roam-insert
+        :desc "org-roam-switch-to-buffer" "b" #'org-roam-switch-to-buffer
+        :desc "org-roam-find-file" "f" #'org-roam-find-file
+        :desc "org-roam-show-graph" "g" #'org-roam-show-graph
+        :desc "org-roam-insert" "i" #'org-roam-insert)
+  (setq org-roam-directory org-directory)
+  :config
+  (require 'org-roam-protocol))
+
+(use-package company-org-roam
```
Comment (zeronone) [inline modules/lang/org/contrib/roam.el]: Is it supposed to be `use-package!`?

**Pull request closed without merging**